### PR TITLE
Multiple winners e2e test

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Implements e2e test for multiple winners per auction feature.

Verifies that:
1. Two winners are selected
2. Winning solutions trade disjoint tokens
3. Two `/settle` calls are sent and two transactions are mined.
4. Two settlements are indexed and dependent data is properly saved to database.

## How to test
Added test.

### CODE REVIEW

**Review by MartinquaXD — APPROVED** [review]

File: crates/e2e/tests/e2e/limit_orders.rs [hunk 1]
```diff
@@ -291,6 +297,147 @@ async fn two_limit_orders_test(web3: Web3) {
...
+    let [token_a, token_b, token_c, token_d] = onchain
+        .deploy_tokens_with_weth_uni_v2_pools(to_wei(1_000), to_wei(1_000))
+        .await;
+
+    // Fund traders
+    token_a.mint(trader_a.address(), to_wei(10)).await;
+    token_b.mint(trader_b.address(), to_wei(10)).await;
+
+    // Create more liquid routes between token_a (token_b) and weth via base_a
+    // (base_b). base_a has more liquidity then base_b, leading to the solver that
```
Comment (squadgazzz) [inline crates/e2e/tests/e2e/limit_orders.rs:314]:
```suggestion
    // (base_b). base_a has more liquidity than base_b, leading to the solver that
```

File: crates/e2e/tests/e2e/limit_orders.rs [hunk 2]
```diff
@@ -291,6 +297,147 @@ async fn two_limit_orders_test(web3: Web3) {
     .unwrap();
 }
 
+async fn two_limit_orders_multiple_winners_test(web3: Web3) {
+    let mut onchain = OnchainComponents::deploy(web3).await;
+
+    let [solver1, solver2] = onchain.make_solvers(to_wei(1)).await;
```
Comment (squadgazzz) [inline crates/e2e/tests/e2e/limit_orders.rs:303]: nit to keep the naming consistent.
```suggestion
    let [solver_a, solver_b] = onchain.make_solvers(to_wei(1)).await;
```

File: crates/e2e/tests/e2e/limit_orders.rs [hunk 3]
```diff
@@ -291,6 +297,147 @@ async fn two_limit_orders_test(web3: Web3) {
...
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader_b.private_key()).unwrap()),
+    );
+    let uid_b = services.create_order(&order_b).await.unwrap();
+
+    // Wait for trade
+    let indexed_trades = || async {
+        onchain.mint_block().await;
+        match services.get_trades(&uid_a).await.unwrap().first() {
```
Comment (squadgazzz) [inline crates/e2e/tests/e2e/limit_orders.rs:417]: Should we also check for the `order_b` here?

File: crates/e2e/src/setup/services.rs [hunk 4]
```diff
@@ -204,6 +204,8 @@ impl<'a> Services<'a> {
                     solver,
                     self.contracts.weth.address(),
                     vec![],
+                    1,
```
[the remark this answers is not in the document]
Comment (sunce86) [reply crates/e2e/src/setup/services.rs:207]: Not yet here, this is where the default value should be used which is still 1.

**Pull request merged**